### PR TITLE
[App] Enable running an app from the Gallery

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR enables running an app directly from the App Gallery (installing + running).

Fixes #\<issue_number>

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

File: src/lightning_app/cli/lightning_cli.py [hunk 1]
```diff
@@ -232,7 +232,12 @@ def _run_app(
     secret: tuple,
     run_app_comment_commands: bool,
 ) -> None:
-    file = _prepare_file(file)
+
+    if not os.path.exists(file):
```
Comment (hhsecond) [inline src/lightning_app/cli/lightning_cli.py:236]: if user makes a mistake in the file name, `_validate_name` from the `gallery_apps_and_components` will throw weird errors. Can do valid name check before this `exists` check and raise nice error message

Comment (tchaton) [reply src/lightning_app/cli/lightning_cli.py:236]: Resolved this by failing later on.

Comment (hhsecond) [reply src/lightning_app/cli/lightning_cli.py:236]: It wouldn't reach "later on" if `_valid_name` function fails. Like try `directory/wrongfile.py`?

File: src/lightning_app/cli/cmd_install.py [hunk 2]
```diff
@@ -14,7 +15,117 @@
...
+def install_component(name: str, yes: bool, version: str) -> None:
+    _install_component_command(name, yes, version)
+
+
+def _install_app_command(name: str, yes: bool, version: str, overwrite: bool = False) -> None:
+    if "github.com" in name:
+        if version != "latest":
+            logger.warn(
+                f"The provided version {version} isn't the officially supported one. "
+                f"The provided version will be ignored."
```
Comment (ethanwharris) [inline src/lightning_app/cli/cmd_install.py:76]: Small wording suggestion:
```suggestion
                "When installing from GitHub, only the 'latest' version is supported. "
                f"The provided version ({version}) will be ignored."
```

File: src/lightning_app/cli/cmd_install.py [hunk 3]
```diff
@@ -14,7 +15,117 @@
...
+    else:
+        return gallery_app(name, yes, version, overwrite=overwrite)
+
+
+def _install_component_command(name: str, yes: bool, version: str, overwrite: bool = False) -> None:
+    if "github.com" in name:
+        if version != "latest":
+            logger.warn(
+                f"The provided version {version} isn't the officially supported one. "
+                f"The provided version will be ignored."
```
Comment (ethanwharris) [inline src/lightning_app/cli/cmd_install.py:88]:
```suggestion
                "When installing from GitHub, only the 'latest' version is supported. "
                f"The provided version ({version}) will be ignored."
```

**Review by ethanwharris — APPROVED** [review]
LGTM, small suggestion

**Review by Borda — APPROVED** [review]
tests... LGTM

**Pull request merged**